pkg/liqo-controller-manager/offloadingStatus-controller/offloadingStatus_controller.go

```diff
@@ -0,0 +1,135 @@
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// OffloadingStatusReconciler checks the status of all remote namespaces associated with this
+// namespaceOffloading Resource, and sets the global offloading status only when there is a feedback
+// from all remote clusters.
+type OffloadingStatusReconciler struct {
+	client.Client
+	Scheme               *runtime.Scheme
+	RemoteClustersNumber int
+}
+
+// Reconcile checks if all remote Namespaces have been created with success or not.
+func (r *OffloadingStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      liqoconst.DefaultNameNamespaceOffloading,
+	}, namespaceOffloading); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no NamespaceOffloading resource in Namespace '%s'", req.Namespace)
+			return ctrl.Result{}, nil
+		}
+		klog.Errorf("%s --> Unable to get namespaceOffloading for the namespace '%s'", err, req.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	if r.RemoteClustersNumber == 0 {
+		virtualNodes := &corev1.NodeList{}
+		if err := r.List(context.TODO(), virtualNodes,
+			client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode}); err != nil {
+			klog.Error(err, " --> Unable to List all virtual nodes")
+			return ctrl.Result{}, err
+		}
+		if len(virtualNodes.Items) == 0 {
+			klog.Info("No VirtualNodes at the moment in the cluster")
+			return ctrl.Result{}, nil
+		}
+		r.RemoteClustersNumber = len(virtualNodes.Items)
+	}
+
+	// if every remote Namespace has a condition
+	if len(namespaceOffloading.Status.RemoteNamespacesConditions) == r.RemoteClustersNumber {
+		ready := 0
+		notReady := 0
+
+		for i := range namespaceOffloading.Status.RemoteNamespacesConditions {
+			condition := liqoutils.GetNamespaceOffloadingCondition(namespaceOffloading.Status.RemoteNamespacesConditions[i],
+				offv1alpha1.NamespaceReady)
+			if condition == nil {
+				continue
+			}
+			if condition.Status == corev1.ConditionTrue {
+				ready++
+			} else {
+				notReady++
+			}
+		}
+
+		switch {
+		case ready == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.AllFailedOffloadingPhaseType
+		case notReady == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.ReadyOffloadingPhaseType
+		default:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.SomeFailedOffloadingPhaseType
+		}
```

Comment:
can we add a phase indicating that for some remote namespace we not have a status yet?

(the case of condition == nil I think)

Reply:
The case of condition == nil is for namespaces that don't have to be offloaded, they have a condition `OffloadingRequired=false`, if some namespaces don't have a status yet the global status OffloadingPhase is still in Progress, i thought this should be enough as first step. What do you think about it ?

Reply:
yes, it is because of the possibility of a new cluster added, the state can change again in pending state, but as before it will be in a future pr I imagine